### PR TITLE
Client receiving and running commands

### PR DESCRIPTION
**Pull request by johanlovgren**
Closing #30, #12 and  #13

### CODE REVIEW

File: client/src/main.rs [hunk 1]
```diff
@@ -1,29 +1,73 @@
...
-// Note that URL is set by environment variable during compilation
-const URL: &'static str = env!("CONTROL_SERVER_URL");
 const POLL_SLEEP: time::Duration = time::Duration::from_secs(5);
 
 
-#[tokio::main]
-async fn main() -> Result<(), anyhow::Error> {
+/// Fetches commands from the control server and runs them
+async fn call_home(token: &String) {
+    let tasks: Vec<Task> = match communication::get_commands(token).await {
```
Comment (Rickebo) [inline client/src/main.rs:17]: Great that the client keeps running even if it fails to get commands!

**Review by Rickebo — APPROVED** [review]
Looks good! Tests run successfully on Windows, and compilation only shows one warning of a line of unused code Gholam already mentioned.

File: client/src/tasks.rs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    io::Write,
+    str,
+    thread,
+    time,
+};
+use rand::Rng;
+use crate::models::TaskResult;
+use crate::Task;
+
+const TERMINAL_CMD: &'static str = "terminal";
```
[the remark this answers is not in the document]
Comment (johanlovgren) [reply client/src/tasks.rs:12]: Yes, this constant is intended to be used as soon as there are more tasks available than "terminal".

File: client/src/communication.rs [hunk 3]
```diff
@@ -1,78 +1,225 @@
...
+const CONTENT_HEADER_VALUE: &'static str = "application/json";
+const AUTHORIZATION_HEADER: &'static str = "Authorization";
+
+/// Sends a post request to the control server with a JSON String as body
+///
+/// # Returns
+/// The response from the request as a String
+///
+/// # Errors
+/// This function fails if there are issues sending the request, if 200 is not returned from
```
[the remark this answers is not in the document]
Comment (johanlovgren) [reply client/src/communication.rs:25]: Yes, that's the case. This just tells that the caller of this function is responsible for handling the errors that this function may raise.

**Pull request merged**